openverse_catalog/dags/database/recreate_popularity_calculation_dag_factory.py

```diff
@@ -0,0 +1,129 @@
+"""
+This file generates Apache Airflow DAGs that, for the given media type,
+completely wipe out the PostgreSQL relations and functions involved in
+calculating our standardized popularity metric. It then recreates relations
+and functions to make the calculation, and performs an initial calculation.
+The results are available in the materialized view for that media type.
+
+These DAGs are not on a schedule, and should only be run manually when new
+SQL code is deployed for the calculation.
+"""
+from airflow import DAG
+from airflow.operators.python import PythonOperator
+from common.constants import DAG_DEFAULT_ARGS, POSTGRES_CONN_ID
+from common.popularity import sql
+from data_refresh.data_refresh_types import DATA_REFRESH_CONFIGS, DataRefresh
+
+
+def create_recreate_popularity_calculation_dag(data_refresh: DataRefresh):
+    media_type = data_refresh.media_type
+    default_args = {
+        **DAG_DEFAULT_ARGS,
+        **data_refresh.default_args,
+    }
+
+    dag = DAG(
+        dag_id=f"recreate_{media_type}_popularity_calculation",
+        default_args=default_args,
+        max_active_tasks=1,
+        max_active_runs=1,
+        schedule_interval=None,
+        catchup=False,
+        doc_md=__doc__,
+        tags=["database", "data_refresh"],
+    )
+    with dag:
+        drop_relations = PythonOperator(
+            task_id="drop_popularity_relations",
+            python_callable=sql.drop_media_popularity_relations,
+            op_args=[POSTGRES_CONN_ID, media_type],
+            doc="Drop the existing popularity views and tables.",
+        )
+
+        drop_functions = PythonOperator(
```

Comment:
This DAG seems like it would be a prime candidate for airflow's [TaskFlow API](https://airflow.apache.org/docs/apache-airflow/stable/tutorial_taskflow_api.html)! Out of the scope of this PR, but something to keep in mind as a good testing ground for trying this feature out.

Reply:
Oh, very cool -- thanks for the link! Yeah this definitely looks like it will work easily here.